### DISCUSSION TITLE
Dead keys break input echo on Linux

### DISCUSSION THREAD

**Discussion by edevil** [ghostty-org/ghostty#2577] [Q&A] [answered]
I'm using the latest master (d47df211b03894427592851b0be3b528ff8221ff) and when I type a dead key then anything else doesn't work.

https://github.com/user-attachments/assets/248a34fa-1829-42e8-8bcd-d6bf776dc50f

I don't have this problem in the Mac version.

**Comment by mitchellh** [comment]
GTK dead keys behave quite differently from macOS, and GTK is responsible for telling us when dead key states start and end. So this is probably just a quirk of the UI subsystem. Can you please get more specific with what you're entering exactly, on what keyboard layout? I can try to reproduce it.

While in a dead key state, input is _not sent to the pty_ until it is _committed_. The UI framework tells us when to commit.

**Comment by edevil** [comment]
```
➜  ~ fastfetch
                             ....              andre@gym
              .',:clooo:  .:looooo:.           ---------
           .;looooooooc  .oooooooooo'          OS: Ubuntu noble 24.04 x86_64
        .;looooool:,''.  :ooooooooooc          Kernel: Linux 6.8.0-48-generic
       ;looool;.         'oooooooooo,          Uptime: 3 hours, 43 mins
      ;clool'             .cooooooc.  ,,       Packages: 3547 (dpkg), 37 (nix-user), 34 (snap)
         ...                ......  .:oo,      Shell: zsh 5.9
  .;clol:,.                        .loooo'     Display (ROG PG279Q): 2560x1440 @ 144 Hz in 27" [External]
 :ooooooooo,                        'ooool     DE: GNOME 46.0
'ooooooooooo.                        loooo.    WM: Mutter (X11)
'ooooooooool                         coooo.    WM Theme: Yaru
 ,loooooooc.                        .loooo.    Theme: Yaru [GTK2/3/4]
   .,;;;'.                          ;ooooc     Icons: Yaru [GTK2/3/4]
       ...                         ,ooool.     Font: Ubuntu (11pt) [GTK2/3/4]
    .cooooc.              ..',,'.  .cooo.      Cursor: Yaru (24px)
      ;ooooo:.           ;oooooooc.  :l.       Terminal: ghostty 0.1.0-main+d47df211
       .coooooc,..      coooooooooo.           CPU: Intel(R) Core(TM) i5-7600K (4) @ 4.40 GHz
         .:ooooooolc:. .ooooooooooo'           GPU: NVIDIA GeForce GTX 1080
           .':loooooo;  ,oooooooooc            Memory: 8.11 GiB / 31.28 GiB (26%)
               ..';::c'  .;loooo:'             Swap: 0 B / 487.00 MiB (0%)
                                               Disk (/): 283.42 GiB / 456.91 GiB (62%) - ext4
                                               Local IP (enp0s31f6): 192.168.100.82/24
                                               Locale: en_US.UTF-8

```
I'm using a Portuguese layout. And the problem manifests itself when I type accents such as: ~, ´, etc.

**Reply by mitchellh** [reply]
Thank you, just because I'm very unfamiliar with the layout: can you tell me what keys on a US layout physical keyboard those accents would align to, if you know?

**Reply by edevil** [reply]
> Thank you, just because I'm very unfamiliar with the layout: can you tell me what keys on a US layout physical keyboard those accents would align to, if you know?

In a US layout these keys would be "right bracket" and "apostrophe".

**Comment by edevil** [comment]
Ghostty keyboard inspector output.

MacOS(works)
![Screenshot 2024-11-05 at 11 46 53](https://github.com/user-attachments/assets/c3f70d1d-97b8-45be-b1ee-f4ac44ec72a9)

Linux(does not work)
![Screenshot from 2024-11-05 22-56-20](https://github.com/user-attachments/assets/0eea05de-25f4-4590-b69b-3230b3f0be73)

**Comment by edevil** [comment, marked as answer]
It turns out it was a [bug](https://github.com/ibus/ibus/issues/2588) in ibus. Ubuntu 24.04 LTS uses version 1.5.29, which has this bug. After upgrading to ubuntu 24.10, which has version 1.5.30, the problem went away.

I don't know why only Ghostty was affected though.

**Reply by nikolaik** [reply]
Had the same issue, thanks for posting your solution <3

**Reply by llynch** [reply]
running ghostty on ubuntu 24.04.1 LTS with ibus version 1.5.29 and gtk version 4.14.2 fixes the accent issue for me.
`GTK_IM_MODULE=simple`

found this tip here: https://bugs.launchpad.net/ubuntu/+source/ibus/+bug/2064025
